src/bio/bio_xstream.c

```diff
@@ -861,6 +849,21 @@ init_blobstore_ctxt(struct bio_xs_context *ctxt, int xs_id)
 		return -DER_NOMEM;
 	}
 
+	/* Writable descriptor required for device health monitoring */
+	rc = spdk_bdev_open(d_bdev->bb_bdev, true, NULL, NULL,
+			    &ctxt->bxc_blobstore->bb_dev_health->bhm_desc);
+	if (rc != 0) {
+		D_ERROR("Failed to open bdev %s, %d\n",
+			spdk_bdev_get_name(d_bdev->bb_bdev), rc);
+		return daos_errno2der(-rc);
+	}
+	health_desc = ctxt->bxc_blobstore->bb_dev_health->bhm_desc;
+
+	/* Get and hold I/O channel for device health monitoring */
+	channel = spdk_bdev_get_io_channel(health_desc);
+	D_ASSERT(channel != NULL);
+	ctxt->bxc_blobstore->bb_dev_health->bhm_io_channel = channel;
+
```

Comment:
This is not correct, above code will be called by every blobstore user xstreams, it should be called by owner xstream only. It should be done in above "if (d_bdev->bb_blobstore == NULL)" branch, that's for device owner only.

Probably move them into alloc_bio_health_monitoring() and rename the function as bio_init_health_monitoring()? The free_bio_health_monitoring() can be renamed as bio_fini_health_monitoring(), and it'll close bdev and put io channel.

Reply:
done